java/client/src/main/java/glide/api/commands/servermodules/GlideJson.java

```diff
@@ -0,0 +1,411 @@
+     *     will be modified only if <code>value</code> is added as the last child in the specified
+     *     <code>path</code>, or if the specified <code>path</code> acts as the parent of a new child
+     *     being added.
+     * @param value The value to set at the specific path, in JSON formatted string.
+     * @param setCondition Set the value only if the given condition is met (within the key or path).
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>set_condition</code>, returns null.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, "doc", , ".", "{'a': 1.0, 'b': 2}", ConditionalChange.ONLY_IF_DOES_NOT_EXIST).get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    public static CompletableFuture<String> set(
+            @NonNull BaseClient client,
+            @NonNull String key,
+            @NonNull String path,
+            @NonNull String value,
+            @NonNull ConditionalChange setCondition) {
+        return executeCommand(
+                        client, new String[] {JSON_SET, key, path, value, setCondition.getValkeyApi()})
+                .thenApply(r -> (String) r);
+    }
+
+    /**
+     * Sets the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path Represents the path within the JSON document where the value will be set. The key
+     *     will be modified only if <code>value</code> is added as the last child in the specified
+     *     <code>path</code>, or if the specified <code>path</code> acts as the parent of a new child
+     *     being added.
+     * @param value The value to set at the specific path, in JSON formatted GlideString.
+     * @param setCondition Set the value only if the given condition is met (within the key or path).
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>set_condition</code>, returns null.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, gs("doc"), , gs("."), gs("{'a': 1.0, 'b': 2}"), ConditionalChange.ONLY_IF_DOES_NOT_EXIST).get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    public static CompletableFuture<String> set(
+            @NonNull BaseClient client,
+            @NonNull GlideString key,
+            @NonNull GlideString path,
+            @NonNull GlideString value,
+            @NonNull ConditionalChange setCondition) {
+        return executeCommand(
+                        client,
+                        new GlideString[] {gs(JSON_SET), key, path, value, gs(setCondition.getValkeyApi())})
+                .thenApply(r -> (String) r);
```

Comment:
I had to remove `thenApply` from `executeCommand` as it would fail the UT, with the <Object> return type. Now if we are changing the return type to <String> then I can move it back.